recipe/build.sh

```diff
@@ -1,40 +1,65 @@
-install -d $PREFIX/epics
 
-# export EPICS_HOST_ARCH as determined
+# We let PREFIX being expanded everywhere in the script.
+# conda will automatically replace it during installation.
+# Note that this isn't true for BUILD_PREFIX.
+EPICS_BASE="$PREFIX/epics"
 EPICS_HOST_ARCH=$(perl src/tools/EpicsHostArch.pl)
-export EPICS_HOST_ARCH
 
-make \
-    CMPLR_PREFIX="${HOST}-" \
-    G++="${CXX}" \
-    GNU_DIR="${BUILD_PREFIX}" \
-    -j ${CPU_COUNT}
+cat << EOF >> configure/CONFIG_SITE.local
+INSTALL_LOCATION=${EPICS_BASE}
+MSI=\$(EPICS_BASE)/bin/\$(EPICS_HOST_ARCH)/msi
+EOF
 
-EPICS_BASE=$PREFIX/epics
+cat << EOF >> configure/os/CONFIG_SITE.Common.linux-x86_64
+# Set GNU_DIR to BUILD_PREFIX or PREFIX if not set (when not using conda-build)
+# Allow to compile without conda-build by installing manually the compilers
+# in a local conda environment
+GNU_DIR = \$(or \$(BUILD_PREFIX),$PREFIX)
+CMPLR_PREFIX = x86_64-conda_cos6-linux-gnu-
 
-# Copy libraries into $PREFIX/lib
-cp -av $PREFIX/epics/lib/$EPICS_HOST_ARCH/lib*${SHLIB_EXT}* $PREFIX/lib
+# --disable-new-dtags is required to avoid LD_LIBRARY_PATH overrride RPATH settings
+OP_SYS_LDFLAGS += -Wl,--disable-new-dtags -Wl,-rpath,${PREFIX}/lib -Wl,-rpath-link,${PREFIX}/lib -L${PREFIX}/lib -Wl,-rpath-link,${EPICS_BASE}/lib/linux-x86_64
+OP_SYS_INCLUDES += -I${PREFIX}/include
+EOF
 
-# deal with env export
-mkdir -p $PREFIX/etc/conda/activate.d
-mkdir -p $PREFIX/etc/conda/deactivate.d
+cat << EOF >> configure/os/CONFIG_SITE.Common.darwin-x86
+GNU_DIR = \$(or \$(BUILD_PREFIX),$PREFIX)
+CMPLR_PREFIX = x86_64-apple-darwin13.4.0-
+CMPLR_CLASS = clang
+CC = x86_64-apple-darwin13.4.0-clang
+CCC = x86_64-apple-darwin13.4.0-clang++
 
-ACTIVATE=$PREFIX/etc/conda/activate.d/epics_base.sh
-DEACTIVATE=$PREFIX/etc/conda/deactivate.d/epics_base.sh
-ETC=$PREFIX/etc
+OP_SYS_CFLAGS += -isysroot ${CONDA_BUILD_SYSROOT} -mmacosx-version-min=${MACOSX_DEPLOYMENT_TARGET}
+OP_SYS_CXXFLAGS += -isysroot ${CONDA_BUILD_SYSROOT} -mmacosx-version-min=${MACOSX_DEPLOYMENT_TARGET}
+OP_SYS_LDFLAGS += -Wl,-rpath,${PREFIX}/lib -L${PREFIX}/lib
+OP_SYS_INCLUDES += -I${PREFIX}/include
+EOF
 
-# set up
-echo "export EPICS_BASE="$EPICS_BASE >> $ACTIVATE
-echo "export EPICS_HOST_ARCH="$EPICS_HOST_ARCH >> $ACTIVATE
+# Compile epics-base
+# Build fails when using -j${CPU_COUNT} (at least with 7.0.2.1)
+make
 
-# tear down
-echo "unset EPICS_BASE" >> $DEACTIVATE
-echo "unset EPICS_HOST_ARCH" >> $DEACTIVATE
+# Create files to set/unset variables when running
+# activate/deactivate
 
-# clean up after self
-unset ACTIVATE
-unset DEACTIVATE
-unset ETC
+# Note that modifying the PATH in deactivate scripts require conda >= 4.7
+mkdir -p $PREFIX/etc/conda/activate.d
+cat <<EOF > $PREFIX/etc/conda/activate.d/epics-base_activate.sh
+export EPICS_BASE="$EPICS_BASE"
+export EPICS_HOST_ARCH="$EPICS_HOST_ARCH"
+export EPICS_BASE_HOST_BIN="${EPICS_BASE}/bin/${EPICS_HOST_ARCH}"
+export PYEPICS_LIBCA="${EPICS_BASE}/lib/${EPICS_HOST_ARCH}/libca${SHLIB_EXT}"
```

Comment:
IMHO this should not be set... You are changing the behavior of a package via another and it can cause unwanted effects.

Reply:
You are probably right. This variable should go in the pyepics recipe if we switch to this package as dependency. I'll remove it.